pkg/virt-controller/watch/vmi/vmi.go

```diff
@@ -1877,12 +1871,10 @@ func (c *Controller) deleteAttachmentPod(vmi *virtv1.VirtualMachineInstance, att
 
 	vmiKey := controller.VirtualMachineInstanceKey(vmi)
 
-	c.podExpectations.ExpectDeletions(vmiKey, []string{controller.PodKey(attachmentPod)})
-	err := c.clientset.CoreV1().Pods(attachmentPod.GetNamespace()).Delete(context.Background(), attachmentPod.Name, v1.DeleteOptions{
+	err := c.deletePod(vmiKey, attachmentPod, v1.DeleteOptions{
```

Comment:
same as above

Reply:
Here it is not as readable as there is the option, I prefer keeping this one.